src/topology/continuous_function/zero_at_infty.lean

```diff
@@ -0,0 +1,318 @@
+vanish at infinity.
+
+You should also extend this typeclass when you extend `zero_at_infty_continuous_map`. -/
+class zero_at_infty_continuous_map_class (F α β : Type*) [topological_space α] [has_zero β]
+  [topological_space β] extends continuous_map_class F α β :=
+(zero_at_infty (f : F) : tendsto f (cocompact α) (𝓝 0))
+
+export zero_at_infty_continuous_map_class (zero_at_infty)
+
+namespace zero_at_infty_continuous_map
+
+section basics
+
+variables [topological_space β] [has_zero β] [zero_at_infty_continuous_map_class F α β]
+
+instance : zero_at_infty_continuous_map_class (α →C₀ β) α β :=
+{ coe := λ f, f.to_fun,
+  coe_injective' := λ f g h, by { obtain ⟨⟨_, _⟩, _⟩ := f, obtain ⟨⟨_, _⟩, _⟩ := g, congr' },
+  map_continuous := λ f, f.continuous_to_fun,
+  zero_at_infty := λ f, f.zero_at_infty' }
+
+/-- Helper instance for when there's too many metavariables to apply `fun_like.has_coe_to_fun`
+directly. -/
+instance : has_coe_to_fun (α →C₀ β) (λ _, α → β) := fun_like.has_coe_to_fun
+
+instance : has_coe_t F (α →C₀ β) :=
+⟨λ f, { to_fun := f, continuous_to_fun := map_continuous f, zero_at_infty' := zero_at_infty f }⟩
+
+@[simp] lemma coe_to_continuous_fun (f : α →C₀ β) : (f.to_continuous_map : α → β) = f := rfl
+
+
+@[ext] lemma ext {f g : α →C₀ β} (h : ∀ x, f x = g x) : f = g := fun_like.ext _ _ h
+
+/-- Copy of a `zero_at_infinity_continuous_map` with a new `to_fun` equal to the old one. Useful
+to fix definitional equalities. -/
+protected def copy (f : α →C₀ β) (f' : α → β) (h : f' = f) : α →C₀ β :=
+{ to_fun := f',
+  continuous_to_fun := by { rw h, exact f.continuous_to_fun },
+  zero_at_infty' := by { simp_rw h, exact f.zero_at_infty' } }
+
+lemma eq_of_empty [is_empty α] (f g : α →C₀ β) : f = g :=
+ext $ is_empty.elim ‹_›
+
+/-- A continuous function on a compact space is automatically a continuous function vanishing at
+infitnity. -/
+@[simps]
+def continuous_map.lift_zero_at_infty [compact_space α] : C(α, β) ≃ (α →C₀ β) :=
+{ to_fun := λ f, { to_fun := f, continuous_to_fun := f.continuous, zero_at_infty' := by simp },
+  inv_fun := λ f, f,
+  left_inv := λ f, by { ext, refl },
+  right_inv := λ f, by { ext, refl } }
+
+/-- A continuous function on a compact space is automatically a continuous function vanishing at
+infitnity. This is not an instance to avoid type class loops. -/
+def zero_at_infty_continuous_map_class.of_compact {G : Type*} [continuous_map_class G α β]
+  [compact_space α] : zero_at_infty_continuous_map_class G α β :=
+{ coe := λ g, g,
+  coe_injective' := λ f g h, fun_like.coe_fn_eq.mp h,
+  map_continuous := map_continuous,
+  zero_at_infty := by simp }
+
+end basics
+
+section algebraic_structure
+
+variables [topological_space β] (x : α)
+
+instance [has_zero β] : has_zero (α →C₀ β) := ⟨⟨0, tendsto_const_nhds⟩⟩
+
+instance [has_zero β] : inhabited (α →C₀ β) := ⟨0⟩
+
+@[simp] lemma coe_zero [has_zero β] : ⇑(0 : α →C₀ β) = 0 := rfl
+lemma zero_apply [has_zero β] : (0 : α →C₀ β) x = 0 := rfl
```

Comment:
I think the `apply` versions of these lemmas are not needed, given the `coe` versions.  Did you ever have a use case when `apply` worked and `coe` didn't?

Reply:
I haven't needed the apply version in this file, but isn't it standard mathlib practice to include both versions? I could see it being quite annoying if someone did need the apply version and it wasn't there. I'm happy to remove it if we would really never need it though.

Reply:
Yeah, going by what shows up when searching `zero_apply` in the docs, it does seem pretty standard to include both. Let's just leave it there.